### PR TITLE
Update auto import flow logic

### PR DESCRIPTION
**Pull request by german-mergestat**
Implementing logic in new pages for repo auto importing.

add-import

manage-imports

manage-default-syncs

### CODE REVIEW

File: ui/src/pages/settings/repo-auto-imports/[importId].tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  useEffect(() => {
+    const property = data?.repoImport?.type === SYNC_REPO_METHOD.GH_ORG ? 'org' : 'user'
+    setName(data?.repoImport?.settings[property])
+
+    const defaultSyncs = data?.repoImport?.settings.defaultSyncTypes
+
+    const list = data?.repoSyncTypes?.nodes.map(st => ({ type: st.type, description: st.description, shortName: st.shortName, checked: defaultSyncs.includes(st.type) }))
+    setSyncsTypesArray(list || [])
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (patrickdevivo) [inline ui/src/pages/settings/repo-auto-imports/[importId].tsx:46]: just curious about this - is there a reason to skip the lint step here?

File: ui/src/pages/settings/repo-auto-imports/[importId].tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    const newSettings = {
+      ...data?.repoImport?.settings,
+      defaultSyncTypes: newDefaultSyncs
+    }
+    updateAutoImport({ variables: { id: importId, settings: newSettings } })
+  }
+
+  const crumbs = [
+    {
+      text: 'Repo Auto imports',
```
Comment (patrickdevivo) [inline ui/src/pages/settings/repo-auto-imports/[importId].tsx:71]:
```suggestion
      text: 'Repo Auto Imports',
```

File: ui/src/views/repositories/modals/remove-import-modal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  const { setShowRemoveImportModal } = useRepositoriesSetState()
+  const { refetch } = useRepoImports()
+
+  const close = useCallback(() => {
+    setShowRemoveImportModal(false)
+  }, [setShowRemoveImportModal])
+
+  const [removeImport] = useMutation(REMOVE_IMPORT, {
+    onCompleted: () => {
+      showSuccessAlert('Import removed successfully')
```
Comment (patrickdevivo) [inline ui/src/views/repositories/modals/remove-import-modal/index.tsx:22]:
```suggestion
      showSuccessAlert('Repo auto import removed')
```

File: ui/src/views/repositories/modals/remove-import-modal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+                onClick={close}
+                startIcon={<XIcon className="t-icon" />}
+              />
+            </Toolbar.Item>
+          </Toolbar.Right>
+        </Toolbar>
+      </Modal.Header>
+      <Modal.Body>
+        <div className='px-6 py-6'>
+          Are you sure you want to remove <b>{importToRemove?.name}</b>?
```
Comment (patrickdevivo) [inline ui/src/views/repositories/modals/remove-import-modal/index.tsx:50]:
```suggestion
          Are you sure you want to remove the GitHub repo auto import for the <b>{importToRemove?.name}</b> {isOrgCheck ? 'organization' : 'user'}?
```

(`isOrgCheck` is made up, would need to be filled in properly)

**Review by patrickdevivo — CHANGES_REQUESTED** [review]

File: ui/src/pages/settings/repo-auto-imports/[importId].tsx (hunk 1 above)
Comment (german-mergestat) [reply ui/src/pages/settings/repo-auto-imports/[importId].tsx]: ohhh no for this case it is no needed. [aebfd74](https://github.com/mergestat/fuse/pull/377/commits/aebfd744874665fe3977e4a3eaffdc7e62402359)

**Review by patrickdevivo — APPROVED** [review]

**Pull request merged**